src/main/java/org/apache/maven/plugins/pmd/PmdReport.java

```diff
@@ -769,7 +769,9 @@ private void configureTypeResolution( PMDConfiguration configuration ) throws Ma
                         for ( String path : projectCompileClasspath )
                         {
                             File pathFile = new File( path );
-                            if ( !pathFile.exists() || pathFile.list().length == 0 )
+                            String[] children = pathFile.list();
+                            
+                            if ( !pathFile.exists() || children == null || children.length == 0 )
```

Comment:
@wcarmon I think, the logic should actually be 

```suggestion
                            if ( !pathFile.exists() || ( children != null && children.length == 0 ) )
```

However, I'm wondering how a jar file can end up in this place. I expected here something like the paths "project-a/target/classes" or "project-a/target/generated-classes" or so.

You've mentioned, you get the null pointer when path is `~/.m2/repository/org/apache/logging/log4j/log4j-api/2.11.1/log4j-api-2.11.1.jar`. Do you use log4j-api as a normal dependency or is there something special in your project?

I've for now squashed and pushed your change to a new branch, let's see what Jenkins says: https://builds.apache.org/view/M-R/view/Maven/job/maven-box/job/maven-pmd-plugin/job/MPMD-288/

Reply:
I've changed it to only consider non-existing files or empty directories - Jenkins is still happy: https://builds.apache.org/view/M-R/view/Maven/job/maven-box/job/maven-pmd-plugin/job/MPMD-288/2/

So, I'll merge this PR, thanks again!